forward/request_sender.go

```diff
@@ -208,13 +236,82 @@ func (s *requestSender) MakeCall(ctx context.Context, res *[]byte, fwdError *err
 			return
 		}
 
-		*res = arg3
 		done <- true
 	}()
 
 	return done
 }
 
+func forwardArgs(call *tchannel.OutboundCall, arg2, arg3 []byte, res *[]byte) error {
+	if err := writeArg(call.Arg2Writer, arg2); err != nil {
+		return err
+	}
+	if err := writeArg(call.Arg3Writer, arg3); err != nil {
+		return err
+	}
+
+	resp := call.Response()
+	if err := discardArg(resp.Arg2Reader); err != nil {
+		return err
+	}
+
+	buf := bufPool.Get()
+	defer bufPool.Put(buf)
+	if err := bufReadArg(buf, resp.Arg3Reader); err != nil {
+		return err
+	}
+
+	if res == nil {
+		*res = []byte{}
+	}
+	*res = append(*res, buf.Bytes()...)
```

Comment:
we don't need the bytes if `res` is `nil`, can do:
```
if res != nil {
  *res = buf.Bytes()
}
return nil
```

Reply:
Derp, I meant to make that a `*res != nil`... pointers to slices are a bit of a change up.

`res` itself will never be nil ( this isn't an exported interface, the only caller is here in the ringpop forwarding package, and it always passes a pointer to a nil slice.... which actually , append will always work... what even am I saying...